modules/auxiliary/gather/zoomeye_search.rb

```diff
@@ -152,27 +140,126 @@ def run
+      while page < maxpage
+        page_result = dork_search(resource, dork, page+1, facets)
+        if page_result['matches'].nil?
+          next
+        end
+        results[page] = page_result
+        page += 1
+      end
+    end
+    tbl1 = Rex::Text::Table.new(
+      'Header'  => 'Search Results',
+      'Indent'  => 1,
+      'Columns' => ['IP:Port', 'City', 'Country', 'Hostname', 'OS', 'Service:Version', 'Info']
+    )
+    tbl2 = Rex::Text::Table.new(
+      'Header'  => 'Search Results',
+      'Indent'  => 1,
+      'Columns' => ['IP', 'Site', 'City', 'Country', 'DB:Version', 'WebApp:Version']
+    )
     # scroll max pages from ZoomEye
-    while page <= maxpage
-      print_status("ZoomEye #{resource} Search: #{dork} - page: #{page}")
-      results = dork_search(dork, resource, page) if dork
-      break unless match_records?(results)
-
-      matches = results['matches']
-      if resource.include?('web')
-        parse_web_records(matches)
+    results.each do |page|
+      if facets
+        fac = page['facets']
+        fac.each do |fa|
+          print_line("#{fa[0]}")
+          fa[1].each do |f|
+            print_line("#{f['name']} count=#{f['count']}")
+          end
+        end
       else
-        parse_host_records(matches)
+        page['matches'].each do |match|
+          city = match['geoinfo']['city']['names']['en']
+          country = match['geoinfo']['country']['names']['en']
+          if resource.include?('host')
+            ip = match['ip']
+            port = match['portinfo']['port']
+            hostname = match['portinfo']['hostname']
+            os = match['portinfo']['os']
+            service = match['portinfo']['app']
+            name = match['portinfo']['name']
+            version = match['portinfo']['version']
+            info = match['portinfo']['extrainfo']
+            report_host(:host     => ip,
+                        :name     => hostname,
+                        :os_name  => os,
+                        :comments => 'Added from Zoomeye'
+                        ) if datastore['DATABASE']
+            report_service(:host => ip,
+                           :port => port,
+                           :proto => name,
+                           :name => "#{service}:#{version}",
+                           :info => info
+                           ) if datastore['DATABASE']
+            tbl1 << ["#{ip}:#{port}", city, country, hostname, os, "#{service}:#{version}", info]
+          else
+            ips = match['ip']
+            site = match['site']
+            database = match['db']
+            dbInfo = []
+            x = 0
+            database.each do |db|
+              dbInfo[x] = "#{db['name']}:#{db['version']}"
+              x += 1
+            end
+            webapp = match['webapp']
+            waInfo = []
+            x = 0
+            webapp.each do |wa|
+              waInfo[x] = "#{wa['name']}:#{wa['version']}"
+              x += 1
+            end
```

Comment:
Same idea here using `#map`.

```suggestion
            waInfo = webapp.map { |wa| "#{wa['name']}:#{wa['version']}" }
```